packages/broker/src/config/ConfigWizard.ts

```diff
@@ -1,254 +1,591 @@
+ */
+async function getPubsubPlugins(): Promise<Partial<Record<PubsubPluginKey, PubsubPlugin>>> {
+    const setupPubsub = await confirm({
+        message:
+            'Do you wish to use your node for data publishing/subscribing?',
+        default: true,
+    })
+
+    if (!setupPubsub) {
+        return {}
+    }
+
+    const keys = await checkbox<PubsubPluginKey>({
+        message: 'Select the plugins to enable',
+        choices: [
+            { value: 'websocket', name: 'WebSocket' },
+            { value: 'mqtt', name: 'MQTT' },
+            { value: 'http', name: 'HTTP' },
+        ],
+    })
+
+    const pubsubPlugins: Awaited<ReturnType<typeof getPubsubPlugins>> = {}
+
+    for (const key of keys) {
+        const defaultPort = DEFAULT_PORTS[key]
 
-export const CONFIG_TEMPLATE: any = {
-    $schema: formSchemaUrl(CURRENT_CONFIGURATION_VERSION),
-    client: {
-        auth: {}
-    },
-    plugins: {},
-    apiAuthentication: {
-        keys: [generateApiKey()]
+        const port = Number(
+            await input({
+                message: `Provide a port for the ${key} plugin`,
+                default: defaultPort.toString(),
+                validate(value) {
+                    try {
+                        return !!z.coerce
+                            .number({
+                                invalid_type_error:
+                                    'Non-numeric value provided',
+                            })
+                            .int('Non-integer value provided')
+                            .min(1024)
+                            .max(49151)
+                            .superRefine((value, ctx) => {
+                                const [pluginKey] =
+                                    Object.entries(pubsubPlugins).find(
+                                        ([pluginKey, plugin]) =>
+                                            value ===
+                                            (plugin.port ||
+                                                DEFAULT_PORTS[
+                                                    pluginKey as PubsubPluginKey
+                                                ])
+                                    ) || []
+
+                                if (pluginKey) {
+                                    ctx.addIssue({
+                                        code: z.ZodIssueCode.custom,
+                                        message: `Port ${value} is taken by ${pluginKey}`,
+                                    })
+                                }
+                            })
+                            .parse(value)
+                    } catch (e: unknown) {
+                        return (e as z.ZodError).issues
+                            .map(({ message }) => message)
+                            .join(', ')
+                    }
+                },
+            })
+        )
+
+        pubsubPlugins[key] = port !== defaultPort ? { port } : {}
```

Comment:
We check if the port has already been taken at line 349. To support that, maybe it would be good to do the omitting of default ports after all the ports have been asked? 

(And it would maybe make sense to omit just the `port` from the object instead of assuming that the port is the only config option we ask. Not too relevant here, but if we add other config options, it is easier if the port omitting feature doesn't give us a surprise by omitting all options.)

Reply:
> We check if the port has already been taken at line 349. To support that, maybe it would be good to do the omitting of default ports after all the ports have been asked?

Not sure if I understand you right, but I already prevent conflicts on both custom and default ports. It's also tested.

> (And it would maybe make sense to omit just the `port` from the object instead of assuming that the port is the only config option we ask. Not too relevant here, but if we add other config options, it is easier if the port omitting feature doesn't give us a surprise by omitting all options.)

Let's be today.